opm/material/fluidsystems/blackoilpvt/LiveOilPvt.hpp

```diff
@@ -579,9 +579,15 @@ class LiveOilPvt
                                                /*temperature=*/Scalar(1e30),
                                                pSat);
 
-            std::pair<Scalar, Scalar> val(Rs, pSat);
+            Pair val(Rs, pSat);
             pSatSamplePoints.push_back(val);
         }
+
+        //Prune duplicate Rs values (can occur, and will cause problems in further interpolation)
+        auto x_coord_comparator = [](const Pair& a, const Pair& b) { return a.first == b.first; };
+        auto last = std::unique(pSatSamplePoints.begin(), pSatSamplePoints.end(), x_coord_comparator);
+        pSatSamplePoints.erase(last, pSatSamplePoints.end());
```

Comment:
I'm not entirely sure how this will work out with the splines, since these are end points and would hence affect more than the interval with zero slope. The issue is that these tables are created based upon evaluations of other tables again. 

I think the actual table is created in ```LiveOilPvt::updateSaturationPressure_(...)```, and if the table is monotonic (but not strictly increasing/decreasing), we may end up with "duplicate" Rs values (four in the case that I was examining) for different pSat values.